src/screens/HomeScreen.tsx

```diff
@@ -67,7 +67,7 @@ type HomeScreenProps = {
 
 export const HomeScreen: React.FC<HomeScreenProps> = ({ navigation }) => {
   const { user } = useAuth()
-  const { secrets, update, remove } = useSecrets()
+  const { secrets, update, remove, updateUserId } = useSecrets()
```

Comment:
This is not a great pattern to use. rather, let's pass the user as an argument to this hook, or alternatively make the useSecrets hook use useAuth. 
If there is no user, the operations are invalid and should throw errors or simply do nothing.
If there is a user everything proceeds as normal

Reply:
Yes, I have changed the code so that useAuth is directly used in Secrets. Please review to see if it looks ok now.